### PR TITLE
Initial push for RSAKey Class

### PR DESCRIPTION
**Pull request by saru95**

### CODE REVIEW

File: CryptoSupport/rsaKeySupport.chpl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+module rsaKeySupport {
+
+  use CryptoUtils;
+  use CryptoUtils;
+  use asymmetricPrimitives;
+  use asymmetricPrimitives;
+
+  proc generateKeys(bits: int) {
+   var localKeyPair: asymmetricPrimitives.EVP_PKEY_PTR;
+   var keyCtx = asymmetricPrimitives.EVP_PKEY_CTX_new_id(6: c_int, c_nil: asymmetricPrimitives.ENGINE_PTR);
```
Comment (panzone) [inline CryptoSupport/rsaKeySupport.chpl]: You should use constants (which should be EVP_PKEY_RSA for PKCS1 if I remember correctly) when possible.

File: CryptoSupport/rsaKeySupport.chpl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+   asymmetricPrimitives.EVP_PKEY_keygen_init(keyCtx);
+   asymmetricPrimitives.EVP_PKEY_CTX_set_rsa_keygen_bits(keyCtx, bits: c_int);
+   asymmetricPrimitives.EVP_PKEY_keygen(keyCtx, localKeyPair);
+   return localKeyPair;
```
Comment (panzone) [inline CryptoSupport/rsaKeySupport.chpl]: Memory leak: you should free keyCtx.

File: Crypto.chpl [hunk 3]
```diff
@@ -120,4 +127,22 @@ module Crypto {
       return keyBuff;
     }
   }
+
+  class RSAKey {
```
Comment (panzone) [inline Crypto.chpl]: Why in Crypto and not in CryptoUtils like CryptoBuffer?

**Comment by saru95** [comment]
@panzone Made the required changes.

**Pull request merged**